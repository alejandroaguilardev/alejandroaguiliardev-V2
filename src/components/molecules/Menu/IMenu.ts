export interface IMenu {
    pages: string[];
    menu: boolean;
    setMenu: (menu: boolean) => void;
}