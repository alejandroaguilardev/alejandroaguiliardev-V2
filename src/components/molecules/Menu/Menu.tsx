import { Toolbar, Box, IconButton } from "@mui/material"
import { Menu as MenuIcon } from '@mui/icons-material';
import { Curriculum, Logotype, Mode, PageRoute } from "../../atoms"
import { IMenu } from "./IMenu";

export const Menu = ({ pages, menu, setMenu }: IMenu) => {

    return (
        <Toolbar
            disableGutters
            sx={{ display: 'flex', justifyContent: 'space-between', mt: { xs: 1, sm: 0.5 }, mb: { xs: 0.5, sm: 0 } }}
        >
            <Box sx={{ display: 'flex', alignItems: 'center' }}>
                <Logotype />
                <Mode />
            </Box>
            <Box sx={{
                display: { xs: 'none', sm: 'flex' },
                alignItems: 'center',
                gap: '2rem',
            }}>
                {pages.map((page: string) => (
                    <PageRoute page={page} key={page} />
                ))}
                <Curriculum sx={{ pl: 8, pr: 8 }} />
            </Box>
            <Box sx={{
                display: { xs: 'block', sm: 'none' },

            }}>
                <IconButton onClick={() => setMenu(!menu)}>
                    <MenuIcon />
                </IconButton>
            </Box>
        </Toolbar >
    )
}
