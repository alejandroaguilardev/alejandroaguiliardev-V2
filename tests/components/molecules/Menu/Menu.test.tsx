import { render, screen } from "@testing-library/react";
import { Menu } from "../../../../src/components/molecules/Menu/Menu";

jest.mock("../../../../src/components/atoms/PageRoute/PageRoute");
jest.mock("../../../../src/components/atoms/Mode/Mode");

describe("Menu Test", () => {
	const pages = ["page 1", "page 2"];
	const menu = true;
	const setMenu = jest.fn();

	test("toMacthSnapshot", () => {
		const { container } = render(
			<Menu pages={pages} menu={menu} setMenu={setMenu} />
		);

		expect(container).toMatchSnapshot();
	});
});
