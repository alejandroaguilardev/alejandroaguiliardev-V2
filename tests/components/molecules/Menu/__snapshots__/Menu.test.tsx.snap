// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Test toMacthSnapshot 1`] = `
<div>
  <div
    class="MuiToolbar-root MuiToolbar-regular css-1ht4ho5-MuiToolbar-root"
  >
    <div
      class="MuiBox-root css-70qvj9"
    >
      <a
        href="/"
      >
        <div
          class="MuiGrid-root css-2t8riz-MuiGrid-root"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2750%27%20height=%2750%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="Logo Alejandro Aguilar"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-s53yhl-MuiTypography-root"
          >
            ALEJANDRO AGUILAR
          </p>
        </div>
      </a>
    </div>
    <div
      class="MuiBox-root css-1vfocle"
    >
      <a
        href="https://drive.google.com/file/d/13xkWuHrDCPpXNNkmPCKqC0BOH6n9nxmo/view?usp=sharing"
        rel="”noopener noreferrer"
        target="_blank"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1a9m0jg-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Get CV
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </a>
    </div>
    <div
      class="MuiBox-root css-10q9jzs"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
